### DISCUSSION TITLE
Conversion of Eigen::Affine types

### DISCUSSION THREAD

**Discussion by Holtskog** [pybind/pybind11#4063] [Q&A] [answered]
Currently it seems like there's no pybind11 support for the Affine data types, namely Affine3d and Affine3f. Could this be implemented in a simple way or are those types simply ignnored?

When I run my code all the conversion works really well until I get to the Affine3d type. I then get this error in python:
`TypeError: Unregistered type : Eigen::Transform<double, 3, 2, 0>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "\<stdin\>", line 1, in \<module\>
TypeError: Unable to convert function return value to a Python type! The signature was
        (self: PyBindModule.WrappedObject) -> Eigen::Transform<double, 3, 2, 0>`

**Comment by Skylion007** [comment, marked as answer]
The issue is what would it be transformed too? If I recall, there are Eigen methods to get the transformation matrices out of the Eigen::Transform object. Returning those transforms as numpy arrays to Python works fine: https://github.com/facebookresearch/habitat-sim/blob/d57275e5739a0c8518e0a742afcfa8945ca75d1c/src/esp/bindings/GeoBindings.cpp#L55
